contracts/sol6/Dao/mock/KyberTokenFeeHandler.sol

```diff
@@ -0,0 +1,582 @@
+pragma solidity 0.6.6;
+
+import "../../utils/Utils5.sol";
+import "../../utils/zeppelin/ReentrancyGuard.sol";
+import "../../utils/zeppelin/SafeERC20.sol";
+import "../../IKyberDao.sol";
+import "../../IKyberFeeHandler.sol";
+import "../../IKyberNetworkProxy.sol";
+import "../../ISimpleKyberProxy.sol";
+import "../../IBurnableToken.sol";
+import "./../ISanityRate.sol";
+import "../../utils/zeppelin/SafeMath.sol";
+import "../DaoOperator.sol";
+
+/**
+ * @title IKyberProxy
+ *  This interface combines two interfaces.
+ *  It is needed since we use one function from each of the interfaces.
+ *
+ */
+interface IKyberProxy is IKyberNetworkProxy, ISimpleKyberProxy {
+    // empty block
+}
+
+
+/**
+ * @title kyberTokenFeeHandler
+ *
+ * @dev kyberTokenFeeHandler works tightly with contracts kyberNetwork and kyberDao.
+ *      Some events are moved to interface, for easier usage
+ * @dev Terminology:
+ *          Epoch - Voting campaign time frame in kyberDao.
+ *              kyberDao voting campaigns are in the scope of epochs.
+ *          BRR - Burn / Reward / Rebate. kyberNetwork fee is used for 3 purposes:
+ *              Burning KNC
+ *              Reward an address that staked knc in kyberStaking contract. AKA - stakers
+ *              Rebate reserves for supporting trades.
+ * @dev Code flow:
+ *      1. Accumulating && claiming Fees. Per trade on kyberNetwork, it calls handleFees() function which
+ *          internally accounts for network & platform fees from the trade. Fee distribution:
+ *              rewards: accumulated per epoch. can be claimed by the kyberDao after epoch is concluded.
+ *              rebates: accumulated per rebate wallet, can be claimed any time.
+ *              Burn: accumulated in the contract. Burned value and interval limited with safe check using
+                    sanity rate.
+ *              Platfrom fee: accumulated per platform wallet, can be claimed any time.
+ *      2. Network Fee distribution: Per epoch kyberFeeHandler contract reads BRR distribution percentage 
+ *          from kyberDao. When the data expires, kyberFeeHandler reads updated values.
+ */
+contract KyberTokenFeeHandler is IKyberFeeHandler, Utils5, DaoOperator, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 internal constant DEFAULT_REWARD_BPS = 3000;
```

Comment:
May be remove the default bps value, and add them as constructor params.
Since it is confused as these values could be different from pre-dao vote result

Reply:
kk

Reply:
hmmm, it exists in the current feeHandler code though. Copy-pasta here... it's a default setting anyways, should be fetching from the DAO

Reply:
lets handle this later.
once we write another fee handler.

anyway we will probably need some changes.
and after all, this is a mock contract